.clang-format

```
@@ -0,0 +1,67 @@
+BasedOnStyle: Google
+AccessModifierOffset: -1
+AlignAfterOpenBracket: DontAlign
+AlignConsecutiveAssignments: None
+AlignOperands: DontAlign
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: Always
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortIfStatementsOnASingleLine: Always
+AllowShortLambdasOnASingleLine: All
+AllowShortLoopsOnASingleLine: true
+AlwaysBreakAfterReturnType: None
+BreakTemplateDeclarations: Yes
+BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterCaseLabel: false
+  AfterClass: false
+  AfterControlStatement: Never
+  AfterEnum: false
+  AfterFunction: false
+  AfterNamespace: false
+  AfterUnion: false
+  BeforeCatch: false
+  BeforeElse: false
+  IndentBraces: false
```

This should result in the same outcome:
```suggestion
BreakBeforeBraces: Attach
```

---

good idea, [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1876/commits/3884817a4647c54423275e2aded95b153e5dda8c)

---

Yeah, the bundled clang-format with clion is too old to support this it turned out (not sure where to force it to use the system one)
```
.clang-format:19:1: error: unknown key 'BreakTemplateDeclarations' BreakTemplateDeclarations: Yes ^~~~~~~~~~~~~~~~~~~~~~~~~ Error reading .clang-format: Invalid argument
```

I've readded the equivalent older (marked as deprecated) option.